src/main.nr

```diff
@@ -7,7 +9,21 @@ fn main(value: u32, hash_value: pub u32){
 }
 
 fn calculate_hash(num: u32) -> u32 {
```

Comment:
Since Noir already does mod when integer overflows. You can do something like this 

```
fn calculate_hash(num: u32) -> u32 {

    let mut hash_value: u32 = 0x811C9DC5;

    // convert num into byte array
    let num_arr: [u8] = (num as Field).to_be_bytes(4);

    // iterate over max possible length
    for i in 0..32 {
        let byte = num_arr[i];
        if (byte != 0) {
            let product = (hash_value * PRIME);
            hash_value = product ^ byte as u32;
        }
    }
    hash_value
}
```